src/symbols/analyzer.rs

```diff
@@ -28,28 +29,54 @@ use crate::ModuleGraph;
 use crate::ModuleParser;
 
 use super::collections::AdditiveOnlyMap;
-use super::collections::LockableRefCell;
 use super::cross_module;
 use super::cross_module::Definition;
-use super::ImportedExports;
 use super::ResolvedSymbolDepEntry;
-use super::TypeTraceDiagnostic;
-use super::TypeTraceDiagnosticKind;
-use super::TypeTraceHandler;
 
-#[derive(Clone)]
-pub struct RootSymbol {
-  specifiers_to_ids: HashMap<ModuleSpecifier, ModuleId>,
-  ids_to_symbols: HashMap<ModuleId, ModuleSymbol>,
+pub struct RootSymbol<'a> {
+  module_graph: &'a ModuleGraph,
+  parser: &'a CapturingModuleParser<'a>,
+  specifiers_to_ids: AdditiveOnlyMap<ModuleSpecifier, ModuleId>,
```

Comment:
It's not ideal performance-wise to store the ModuleId boxed here (internally in AdditiveOnlyMap). I will fix this in a future PR.